pkg/util/azureclient/mgmt/authorization/denyassignment_addons.go

```diff
@@ -30,3 +39,71 @@ func (c *denyAssignmentClient) ListForResourceGroup(ctx context.Context, resourc
 
 	return result, nil
 }
+
+// getCreateRequest creates the deny assignment get request.
+func (c *denyAssignmentClient) getDenyAssignmentRequest(ctx context.Context, client DenyAssignmentsARMClient, subscriptionDoc *api.SubscriptionDocument, doc *api.OpenShiftClusterDocument) (*policy.Request, error) {
```

Comment:
Even though the sdk's client doesn't have a delete method, could we still use it's get method?

https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v63.1.0+incompatible/services/preview/authorization/mgmt/2018-09-01-preview/authorization#DenyAssignmentsClient.Get

Reply:
Actually, is this get call before the delete call even needed? What's it for?

Reply:
It looks like I was wrong about this. We randomly generate a name (guid) for the deny assignment using this ARM function: https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#guid and we don't store it anywhere, so we do need a Get to determine the deny assignment name

Reply:
I removed the custom `Get` and replaced it with our custom `ListForResourceGroup`. Let me know what you think!

Reply:
Yeah, that works too! Just wanted to find a way to remove as much custom stuff as we could.